### PR TITLE
hibernate-user-service

### PR DESCRIPTION
**Pull request by etezaadi**
Created model - User;
Implemented UserDao;
Implemented UserService;
Implemented AuthenticationService;
Created two exceptions - RegistrationException and AuthenticationException.

### CODE REVIEW

File: src/main/java/mate/academy/service/impl/AuthenticationServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        Optional<User> userByEmail = userService.findByEmail(email);
+        if (userByEmail.isEmpty() || !userByEmail.get().getPassword()
+                .equals(HashUtil.hashPassword(password, userByEmail.get().getSalt()))) {
+            throw new AuthenticationException("There is no user with this email or password.");
```
Comment (Nick97-git) [inline src/main/java/mate/academy/service/impl/AuthenticationServiceImpl.java:23]:
```suggestion
            throw new AuthenticationException("There is no such user with email: " + email);
```

File: src/main/java/mate/academy/service/impl/AuthenticationServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+    @Override
+    public User register(String email, String password) throws RegistrationException {
+        if (userService.findByEmail(email).isPresent()) {
+            throw new RegistrationException("User with this email address "
+                    + "already exists: " + email);
+        }
+        if (!email.contains("@")) {
+            throw new RegistrationException("Incorrect email address");
+        }
```
Comment (Nick97-git) [inline src/main/java/mate/academy/service/impl/AuthenticationServiceImpl.java:36]: let's remove this check, in one of the next tasks you will be implementing validation logic for email

File: src/main/java/mate/academy/util/HashUtil.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        StringBuilder hashedPassword = new StringBuilder();
+        try {
+            MessageDigest messageDigest = MessageDigest.getInstance(CRYPTO_ALGORITHM);
+            messageDigest.update(salt);
+            byte[] digest = messageDigest.digest(password.getBytes());
+            for (byte b : digest) {
+                hashedPassword.append(String.format("%02x", b));
+            }
+        } catch (NoSuchAlgorithmException e) {
+            throw new IllegalStateException("Can't create hash using SHA-512 algorithm.", e);
```
Comment (Nick97-git) [inline src/main/java/mate/academy/util/HashUtil.java:30]: use constant in the message

**Review by Nick97-git — CHANGES_REQUESTED** [review]

File: src/main/java/mate/academy/service/impl/AuthenticationServiceImpl.java (hunk 1 above)
Comment (etezaadi) [reply src/main/java/mate/academy/service/impl/AuthenticationServiceImpl.java:23]: @Nick97-git i don't think, that it will be secure

Comment (Nick97-git) [reply src/main/java/mate/academy/service/impl/AuthenticationServiceImpl.java:23]: if you'd add password yes, but without it, it's ok

**Review by Nick97-git — CHANGES_REQUESTED** [review]

**Review by Nick97-git — APPROVED** [review]

**Pull request closed without merging**